renku/core/commands/init.py

```diff
@@ -153,18 +161,118 @@ def select_template_from_manifest(
+                value,
+            )
+    else:
+        raise errors.InvalidTemplateError(
+            f"Template contains variable {name} of type {variable_type} which is not supported."
+        )
+
+    if not valid:
+        return False, f"Value '{value}' is not of type {variable_type} required by {name}.", value
+
+    return True, None, value
+
+
+def prompt_for_value(name, template_variable):
+    """Prompt the user for a template value."""
+    valid = False
+    while not valid:
+        variable_type = f', type: {template_variable["type"]}' if "type" in template_variable else ""
+        enum_values = f', options: {template_variable["enum"]}' if "enum" in template_variable else ""
+        default_value = template_variable["default_value"] if "default_value" in template_variable else ""
+
+        value = communication.prompt(
+            msg=(
+                f'The template requires a value for "{name}" ({template_variable["description"]}'
+                f"{variable_type}{enum_values})"
+            ),
+            default=default_value,
+            show_default=bool(default_value),
+        )
+
+        valid, msg, value = validate_template_variable_value(name, template_variable, value)
+
+        if msg:
+            communication.info(msg)
+
+    return value
+
+
 def verify_template_variables(template_data, metadata):
     """Verifies that template variables are correctly set."""
     template_variables = template_data.get("variables", {})
     template_variables_keys = set(template_variables.keys())
     input_parameters_keys = set(metadata.keys())
+
+    for key in template_variables:
+        if "description" not in template_variables[key]:
+            raise errors.InvalidTemplateError(f"Template parameter {key} does not contain a description.")
+
+    for key in sorted(template_variables_keys.intersection(input_parameters_keys)):
+        valid, msg, metadata[key] = validate_template_variable_value(key, template_variables[key], metadata[key])
+
+        if not valid:
+            communication.info(msg)
+
+            metadata[key] = prompt_for_value(key, template_variables[key])
+
     for key in template_variables_keys - input_parameters_keys:
-        value = communication.prompt(
-            msg=f'The template requires a value for "{key}" ({template_variables[key]})',
-            default="",
-            show_default=False,
-        )
-        metadata[key] = value
+        if "default_value" in template_variables[key]:
+            default_value = template_variables[key]["default_value"]
+
+            if not validate_template_variable_value(key, template_variables[key], default_value)[0]:
+                raise errors.InvalidTemplateError(
+                    f"Default value {default_value} for template variable {key} does not match variable type."
+                )
+
+            metadata[key] = default_value
+            continue
```

Comment:
This won't prompt for parameters with a default value. Shouldn't we prompt for a value and use the default value when user enters nothing?